src/data/equiv/transfer_instance.lean

```diff
@@ -124,6 +124,14 @@ protected def semigroup [semigroup β] : semigroup α :=
 let mul := e.has_mul in
 by resetI; apply e.injective.semigroup _; intros; exact e.apply_symm_apply _
 
+/-- Transfer `semigroup_with_zero` across an `equiv` -/
+protected def semigroup_with_zero [semigroup_with_zero β] : semigroup_with_zero α :=
+{ zero_mul := by simp [mul_def, zero_def],
+  mul_zero := by simp [mul_def, zero_def],
+  ..equiv.has_zero e,
+  ..equiv.has_mul e,
+  ..equiv.semigroup e }
```

Comment:
This no longer matches the style of the rest of the proofs in this file. I guess it should be something like

```suggestion
let mul := e.has_mul, zero := e.has_zero in
by resetI; apply e.injective.semigroup_with_zero _; intros; exact e.apply_symm_apply _
```